### PR TITLE
Add namedBuilders for special input encoding of types

### PR DESCRIPTION
**Pull request by mandarjog**
`istio.policy.v1beta1.Value` type is used to denote `interface{}` type in proto.
It has special (simpler) input mapping. 

`valueField: myattribute` automatically gets encoded as the correct oneoff field.
1. Update test proto to include `istio.policy.v1beta1.Value`
2. Update builder and encoder to process the above type.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -179,46 +179,6 @@ jobs:
       - store_test_results:
           path: /go/out/tests
 
-  e2e-mixer:
```
Comment (guptasu) [inline .circleci/config.yml]: why circleci change ?

File: mixer/pkg/protobuf/yaml/dynamic/encoder_test.go [hunk 2]
```diff
@@ -702,17 +718,17 @@ func testMsg(t *testing.T, input string, output string, res protoyaml.Resolver,
...
+	t.Logf("ff2 = %v", ff2)
 
 	// confirm that codegen'd code direct unmarshal and unmarhal thru bytes yields the same result.
 
 	if !reflect.DeepEqual(ff2, ff1) {
 		s, _ := diff.PrettyDiff(ff2, ff1)
 		t.Logf("difference: %s", s)
 		t.Fatalf("\n got: %v\nwant: %v", ff2, ff1)
+	} else {
+		t.Logf("\n got: %v\nwant: %v", ff2, ff1)
```
Comment (guptasu) [inline mixer/pkg/protobuf/yaml/dynamic/encoder_test.go]: why are we printing in case of success

Comment (mandarjog) [reply mixer/pkg/protobuf/yaml/dynamic/encoder_test.go]: I left it there so I see success cases, but it should not be printed for success.
I will remove it later.

**Pull request merged**